apps/ui/src/stores/votingPowers.ts

```diff
@@ -0,0 +1,99 @@
+import { utils } from '@snapshot-labs/sx';
+import { defineStore } from 'pinia';
+import { getNetwork } from '@/networks';
+import { VotingPower, VotingPowerStatus } from '@/networks/types';
+import { Proposal, Space } from '@/types';
+
+const LATEST_BLOCK_NAME = 'latest';
+
+type VotingPowerItem = {
+  votingPowers: VotingPower[];
+  totalVotingPower: bigint;
+  status: VotingPowerStatus;
+  symbol: string;
+  decimals: number;
+  error: utils.errors.VotingPowerDetailsError | null;
+};
+
+export const useVotingPowersStore = defineStore('votingPowers', () => {
+  const votingPowers = reactive<Map<string, VotingPowerItem>>(new Map());
+
+  function getIndex(space: Space, block: number | null): string {
+    return `${space.id}:${block ?? LATEST_BLOCK_NAME}`;
+  }
+
+  function get(
+    space: Space,
+    block: number | null
+  ): VotingPowerItem | undefined {
+    return votingPowers.get(getIndex(space, block));
+  }
+
+  async function fetch(
+    item: Space | Proposal,
+    account: string,
+    block: number | null
+  ) {
+    const space: Space = 'space' in item ? (item.space as Space) : item;
```

Comment:
Could we improve it a bit and avoid casting it to `Space` (`Proposal['space']` is not really equal to `Space`).

We could do something like this, making sure types stay accurate:
```diff
diff --git a/apps/ui/src/stores/votingPowers.ts b/apps/ui/src/stores/votingPowers.ts
index b036f1bc..3124c4bd 100644
--- a/apps/ui/src/stores/votingPowers.ts
+++ b/apps/ui/src/stores/votingPowers.ts
@@ -6,6 +6,7 @@ import { Proposal, Space } from '@/types';
 
 const LATEST_BLOCK_NAME = 'latest';
 
+type SpaceDetails = Proposal['space'];
 type VotingPowerItem = {
   votingPowers: VotingPower[];
   totalVotingPower: bigint;
@@ -18,12 +19,12 @@ type VotingPowerItem = {
 export const useVotingPowersStore = defineStore('votingPowers', () => {
   const votingPowers = reactive<Map<string, VotingPowerItem>>(new Map());
 
-  function getIndex(space: Space, block: number | null): string {
+  function getIndex(space: SpaceDetails, block: number | null): string {
     return `${space.id}:${block ?? LATEST_BLOCK_NAME}`;
   }
 
   function get(
-    space: Space,
+    space: SpaceDetails,
     block: number | null
   ): VotingPowerItem | undefined {
     return votingPowers.get(getIndex(space, block));
@@ -34,7 +35,7 @@ export const useVotingPowersStore = defineStore('votingPowers', () => {
     account: string,
     block: number | null
   ) {
-    const space: Space = 'space' in item ? (item.space as Space) : item;
+    const space = 'space' in item ? item.space : item;
 
     const existingVotingPower = get(space, block);
     if (existingVotingPower && existingVotingPower.status === 'success') return;
```